### DISCUSSION TITLE
Accessing element geometry using front-end/common libs

### DISCUSSION THREAD

**Discussion by kardoka** [iTwin/itwinjs-core#2411] [Q&A] [answered]
Greetings, I want to be able to extract the geometry of elements using the supplied front-end/common libraries.

To the best of my knowledge, using back-end libraries you can obtain a list of [GeometricElement](https://www.itwinjs.org/reference/imodeljs-backend/elements/geometricelement/) objects. From this list you can access the `geom` properties of elements, where the geometry is stored. But this approach would not be possible without the back-end libraries.

What are my options for accessing elements' geometry data using front-end/common libraries? If there are options, where should I start?

EDIT: grammar.

**Comment by bbastings** [comment, marked as answer]
You can use IModelConnection.Elements.loadProps. Request the geometry using ElementLoadProps.wantGeometry.

https://www.itwinjs.org/reference/imodeljs-frontend/imodelconnection/imodelconnection.elements/loadprops/

**Reply by MichaelBelousov** [reply]
new link: https://www.itwinjs.org/reference/core-frontend/imodelconnection/imodelconnection/imodelconnection.elements/loadprops/
